docs/core-concepts/atomic-swap-htlc.md

```diff
@@ -0,0 +1,146 @@
+
+## Hash Time Locked Contract (HTLC)
+
+HTLC stands for Hash Time Locked Contract. An HTLC is a script or smart contract that locks a digital asset over time.
+HTLCs require a ledger that provides a time lock.
+
+Both parties are required to have the same knowledge of all parameters for the HTLCs on both ledgers before they can start.
+Both parties then have the same functions available within the HTLCs and can decide to move forward or wait to take their money back. 
+No party has access to both assets at the same time within the time boundaries of the atomic swap. 
+
+HTLCs typically define three functions:
+
+1. **Fund**: Locking up the asset in the HTLC.
+1. **Redeem**: Until the expiry time-stamp is reached the other party has the chance to take the locked-up assets with the secret.
+1. **Refund**: After the expiry condition has been met, the party that locked up the asset can take it back.
+
+HTLCs are secured by a time-lock and hash-lock.
+The hash-lock requires the generation of a secret prior to funding. 
+The order of the actions is then tied to the cryptographic roles of Alice Bob. 
+Alice is the role that comes up with the secret.
+She shares the hash of that secret with Bob enabling him to fund on his side.
+The party in the cryptographic role of Alice has to fund and redeem first, because she has the actual secret.
+
+## Atomic Swaps using HTLCs in COMIT
+
+Prior to the actual swap execution there is usually a [Negotiation](negotiation.md) phase where the two parties agree on the ledgers, assets and amounts to be swapped.
+
+Assuming that Alice has ETH on Ethereum and Bob has BTC on Bitcoin, Alice wants Bob's BTC and Bob wants Alice's ETH, the execution of a swap would be like this:
+
+1. Alice generates a secret and hashes it. She sends to Bob:
+    1. Hash of the secret. 
+    1. Her refund address on Ethereum.
+    1. Her redeem address on Bitcoin.
+    1. Proposed expiry times for the HTLCs.
+1. Bob receives the swap proposed by Alice and sends back:
+    1. His refund address on Bitcoin.
+    1. His redeem address on Ethereum.
+1. Alice has to fund first, she deploys an Ethereum HTLC transferring the agreed amount of ETH into the contract.
+1. Bob monitors the Ethereum ledger and notices that Alice has deployed the HTLC funding the ETH. 
+He now sends a Bitcoin transaction into the Bitcoin network that is guarded by the script-hash of the Bitcoin HTLC (Bitcoin script).
+1. Alice monitors the Bitcoin ledger and notices that Bob has sent the funding transaction. She can now redeem the Bitcoin by spending the Bitcoin by revealing the secret.
+1. Bob is monitoring the Bitcoin ledger and notices that Alice has spent from the funding transaction by revealing the secret.
+He extracts the actual secret from Alice's redeem transaction and uses it to redeem on Ethereum. 
+
+The COMIT protocol acts as enabler for atomic swaps in applications.
```

Comment:
This could have its own section heading e.g. 'Atomic Swaps using HTLCs in COMIT'.